incubator/exporter-stdout/src/test/java/io/aklivity/zilla/runtime/exporter/stdout/internal/StdoutExporterHandlerTest.java

```diff
@@ -0,0 +1,188 @@
+        layout.writeEvent(MQTT_TYPE_ID, mqttAuthorizationEvent.buffer(), 0, mqttAuthorizationEvent.sizeof());
+        SchemaRegistryEventFW schemaRegistryAuthorizationEvent = new SchemaRegistryEventFW.Builder()
+            .wrap(eventBuffer, 0, eventBuffer.capacity())
+            .remoteAccessRejected(e -> e.timestamp(77)
+                .traceId(0x0000000000000042L)
+                .namespacedId(0x0000000200000007L)
+                .url("url")
+                .method("method")
+                .status((short) 42)
+            ).build();
+        layout.writeEvent(SCHEMA_REGISTRY_TYPE_ID, schemaRegistryAuthorizationEvent.buffer(), 0,
+            schemaRegistryAuthorizationEvent.sizeof());
+        TcpEventFW tcpDnsResolutionFailedEvent = new TcpEventFW.Builder()
+            .wrap(eventBuffer, 0, eventBuffer.capacity())
+            .dnsResolutionFailed(e -> e.timestamp(77)
+                .traceId(0x0000000000000042L)
+                .namespacedId(0x0000000200000007L)
+                .address("address")
+            ).build();
+        layout.writeEvent(TCP_TYPE_ID, tcpDnsResolutionFailedEvent.buffer(), 0, tcpDnsResolutionFailedEvent.sizeof());
+        TlsEventFW tlsFailedEvent = new TlsEventFW.Builder()
+            .wrap(eventBuffer, 0, eventBuffer.capacity())
+            .tlsFailed(e -> e.timestamp(77)
+                .traceId(0x0000000000000042L)
+                .namespacedId(0x0000000200000007L)
+                .error(t -> t.set(TlsError.HANDSHAKE_ERROR))
+            ).build();
+        layout.writeEvent(TLS_TYPE_ID, tlsFailedEvent.buffer(), 0, tlsFailedEvent.sizeof());
+
+        EngineConfiguration config = mock(EngineConfiguration.class);
+        EngineContext context = mock(EngineContext.class);
+        StdoutExporterConfig exporter = new StdoutExporterConfig(mock(ExporterConfig.class));
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        PrintStream ps = new PrintStream(os);
+        when(context.lookupLabelId("http")).thenReturn(HTTP_TYPE_ID);
+        when(context.lookupLabelId("kafka")).thenReturn(KAFKA_TYPE_ID);
+        when(context.lookupLabelId("mqtt")).thenReturn(MQTT_TYPE_ID);
+        when(context.lookupLabelId("schema-registry")).thenReturn(SCHEMA_REGISTRY_TYPE_ID);
+        when(context.lookupLabelId("tcp")).thenReturn(TCP_TYPE_ID);
+        when(context.lookupLabelId("tls")).thenReturn(TLS_TYPE_ID);
+        when(context.supplyNamespace(0x0000000200000007L)).thenReturn("ns");
+        when(context.supplyLocalName(0x0000000200000007L)).thenReturn("binding");
+        when(context.supplyEventReader()).thenReturn(() -> layout::readEvent);
+        StdoutExporterHandler handler = new StdoutExporterHandler(config, context, exporter, ps);
+
+        // WHEN
+        handler.start();
+        for (int i = 0; i < 42; i++)
+        {
+            handler.export();
+        }
+        handler.stop();
+
+        // THEN
+        assertThat(os.toString(), equalTo(EXPECTED_OUTPUT));
```

Comment:
Please avoid these kinds of unit tests as they are too brittle and cluttered.

Instead, please use an IT to drive the behavior you want via the actual bindings and verify the logged event output. The spec scripts yout need should mostly already exist in the corresponding binding spec projects.

Please use separate ITs for each different binding type. Catalog testing might be slightly different for `schema-registry`, but we'd still need an IT script to simulate failure response from the remote schema registry http server.

Note: consider providing `StdoutConfiguration` to override the destination of the output instead of stdout during the IT, so that the output can be easily read and compared with the expected output.

Reply:
I understand this is an eyesore, but IMO it serves a useful purpose. These are the pro arguments:
- it's less than 200 lines, runs in less than half a second, but bumps the coverage of this module from 5% to 95%
- more importantly, it tests the functionality of this module in isolation, making sure that a test failure means a regression here in this module

I understand an integration test would also make sense, to make sure the whole logging workflow is operating as it should be. Yep, I think as as first step I will need to add an option to the `StdoutConfiguration` so we can print the output to a file instead of stdout.